### ISSUE TITLE
i have the below error while executing the commands :( any help

### ISSUE THREAD

**Issue by denver2020** [tuxd3v/ats#6]
[putty - Copy.log](https://github.com/tuxd3v/ats/files/2719763/putty.-.Copy.log)

**Comment by tuxd3v** [comment]
What is the output of:
`ats -t` 
?

If there is any error SysFs related, you need to install at least kernel version 4.4.138-1094..
Do this:
`apt-get install linux-image-4.4.138-1097-rockchip-ayufan-gb5128c0a1684 linux-headers-4.4.138-1097-rockchip-ayufan-gb5128c0a1684 linux-firmware-image-4.4.138-1097-rockchip-ayufan-gb5128c0a1684` 
`reboot`

You should see fan starting at boot..

If you want you can then run also in test mode :)

**Comment by denver2020** [comment]
thanks so much 👍 
on rebooting the fan started spinning and stopped after a few seconds. does that mean the fan will now spin depending on the cpu temp? attached below the log file
[fan test.txt](https://github.com/tuxd3v/ats/files/2720562/fan.test.txt)

**Comment by tuxd3v** [comment]
Yup :)
You Welcome!

Yes the Fan is time trigered, and for each time its triggered, it works more time or less acordingly the Temperatures..
The power that is aplied, is the one provided to you:
`info:'Pratio' timers
info:    'Pratio[ -20 - 40 [' = 0
info:    'Pratio[ 40 ]'       = 40
info:    'Pratio[ 41 ]'       = 50
info:    'Pratio[ 42 ]'       = 61
info:    'Pratio[ 43 ]'       = 72
info:    'Pratio[ 44 ]'       = 83
info:    'Pratio[ 45 ]'       = 93
info:    'Pratio[ 46 ]'       = 104
info:    'Pratio[ 47 ]'       = 115
info:    'Pratio[ 48 ]'       = 126
info:    'Pratio[ 49 ]'       = 136
info:    'Pratio[ 50 ]'       = 147
info:    'Pratio[ 51 ]'       = 158
info:    'Pratio[ 52 ]'       = 169
info:    'Pratio[ 53 ]'       = 179
info:    'Pratio[ 54 ]'       = 190
info:    'Pratio[ 55 ]'       = 201
info:    'Pratio[ 56 ]'       = 212
info:    'Pratio[ 57 ]'       = 222
info:    'Pratio[ 58 ]'       = 233
info:    'Pratio[ 59 ]'       = 244
info:    'Pratio[ 60 ]'       = 255
info:    'Pratio[ 60 - 70 ['  = 255`

For example at 50 Degrees it will have a power applied to Fan of 147, or ( 147 / 255 ) * 100 = 57.65% of total power for Fan.
You can adapt if you want the best power levels, adjusting:
`MAX_CONTINUOUS_THERMAL_TEMP
MIN_CONTINUOUS_THERMAL_TEMP`

in /etc/ats.conf

**Comment by denver2020** [comment]
hi
yes it does . no issues so far

**Comment by tuxd3v** [comment]
Hello,
Thank you for your report!
My knowledge so far is that it should run in majority of systemd based Linux distros..
For SysVinit, we need yet to have a port of a OS to RockPro64( I am planning for Devuan Ascii also, but we need yet a ported first to test.. so that people using it can enjoy an Active Thermal Solution ..)

Any way, thanks for your report :)

**Comment by YuriyGavrilov** [comment]
done. working but with some magic and two reboots. 
armhf image version much more better and stable. Thx.

**Comment by tuxd3v** [comment]
Hello YuriyGavrilov,
You welcome thanks!

The last release version, still compiles for armv8..
I am in the process to improve Makefile, so that it work with several ARM Architectures..
For that,
I am creating a tool to detect arch, or arch clusters like 'BIG.little', automatically, without relying in a lot of MakeFile workflow..
See this:
[tools](https://github.com/tuxd3v/tools/blob/master/aarch)

Long Story short,
I am in the final preparation to have the Application Working on any ARM CPU, from last decade,
with 2 possibilities for install:
1) Luarocks
2) Manually by Makefile

It would be nice to know the problems you encountered, when installing in armhf,( if you can describe something.. I am trying to support  armhf ) 
:)

Thanks in Advance
